Implement Market Home V2 with new filtering options and UI components

limichange opened the pull request:
- Introduced MarketHomeV2 component to enhance market viewing experience.
- Added DiscoveryFilterControl and LiquidityFilterControl for improved filtering capabilities.
- Implemented TimeRangeSelector for time-based data filtering.
- Created MarketTokenList to display market tokens with relevant data.
- Integrated RiskIndicatorCard to provide risk assessment for tokens.
- Added support for network selection with MarketTokenListNetworkSelector.
- Enhanced UI with new components for better user interaction and experience.
- Updated DevSettings to enable/disable Market V2 feature.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a new Market V2 home page with enhanced filtering, sorting, and network selection capabilities.
  - Added Discovery, Liquidity, and Custom filter controls for market tokens.
  - Implemented a paginated, detailed market token table with advanced columns and transaction breakdowns.
  - Provided a risk indicator card component for visualizing token risk levels.
  - Enabled developer toggle to switch between classic and new Market V2 views.

- **Improvements**
  - Updated developer settings to include a toggle for enabling the new Market V2 experience.
  - Enhanced network selection with a searchable network panel and improved UI for selecting blockchain networks.

- **Style/Chores**
  - Various UI and style refinements for a more responsive and user-friendly market browsing experience.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
Implement Market Home V2 with new filtering options and UI components

- Introduced MarketHomeV2 component to enhance market viewing experience.
- Added DiscoveryFilterControl and LiquidityFilterControl for improved filtering capabilities.
- Implemented TimeRangeSelector for time-based data filtering.
- Created MarketTokenList to display market tokens with relevant data.
- Integrated RiskIndicatorCard to provide risk assessment for tokens.
- Added support for network selection with MarketTokenListNetworkSelector.
- Enhanced UI with new components for better user interaction and experience.
- Updated DevSettings to enable/disable Market V2 feature.

limichange committed:
Merge branch 'x' into market-home-v2-ui

limichange committed:
Refactor MarketTokenList to use structured wallet info

- Updated walletInfo in IMarketToken to use ITxnsWalletInfo interface for better data structure.
- Modified Txns component to display wallet info as separate buy and sell counts.
- Removed deprecated walletInfo rendering from MarketTokenColumns.
- Adjusted default market token data to reflect new wallet info format.

limichange committed:
feat: Enhance MarketTokenList with pagination and additional tokens

- Added pagination functionality to the MarketTokenList component for improved navigation.
- Introduced new tokens to the default market token data for demonstration purposes.
- Updated the component to display paginated data based on the current page and page size.

limichange committed:
feat: Enhance RiskIndicatorCard with title color configuration

- Added titleColor property to typeConfig for each risk type to allow dynamic title coloring.
- Updated RiskIndicatorCard component to apply the titleColor based on the risk type.
- Adjusted layout with additional padding and margin for improved visual separation.

limichange committed:
Merge branch 'x' into market-home-v2-ui

limichange committed:
refactor: Update RiskIndicatorCard layout and styling

- Replaced 'space' prop with 'gap' for better alignment in the RiskIndicatorCard component.
- Simplified Icon component rendering by removing unnecessary line breaks for cleaner code.

limichange committed:
refactor: Replace Stack with ScrollView in MarketHomeV2 for improved layout

- Changed the Stack component to a ScrollView to allow for vertical scrolling of the MarketTokenList.
- Updated contentContainerStyle to ensure proper flex behavior and removed vertical scroll indicators for a cleaner UI.

limichange committed:
Merge branch 'x' into market-home-v2-ui

limichange committed:
refactor: Update MarketHomeV2 components for improved functionality and structure

- Commented out the RiskIndicatorCardDemo component in MarketHomeV2 for potential future use.
- Renamed 'defaultData' to 'data' in MarketTokenData and expanded the default data array for better demonstration.
- Increased pageSize from 5 to 10 in MarketTokenList to enhance pagination experience.
- Removed unused Stack import in RiskIndicatorCardDemo for cleaner code.

limichange committed:
Merge branch 'x' into market-home-v2-ui

limichange commented:
@coderabbitai stop and resolve.

limichange committed:
refactor: Standardize filePath prop indentation in gallery components

- Adjusted the indentation of the filePath prop in Icon, JotaiContext, and ListView gallery components for consistent formatting.
- Updated DiscoveryFilterControl and MarketFilterBar components to replace EFilterOption enum with a string union type for improved type safety and simplicity.

limichange committed:
refactor: Update DiscoveryFilterControl and MarketFilterBar for improved type safety

- Replaced string union type with EFilterOption enum in DiscoveryFilterControl and MarketFilterBar components for better type safety.
- Adjusted the import statements accordingly to reflect the changes in type definitions.
- Updated default values and state management to utilize the new enum structure.

limichange committed:
Merge branch 'x' into market-home-v2-ui

limichange commented:
@coderabbitai stop and resolve.

originalix committed:
Merge branch 'x' into market-home-v2-ui

sidmorizon commented on packages/kit/src/views/Market/MarketHomeV2/components/MarketTokenListNetworkSelector/data.ts:
```diff
@@ -0,0 +1,147 @@
+export const swapNetworksIncludeAllNetwork = [
```
直接 import  presetNetworks 呢

presetNetworksMap.eth
presetNetworksMap.btc

limichange committed:
Merge branch 'x' into market-home-v2-ui

limichange committed:
refactor: simplify MarketHomeV2 layout and enhance MarketTokenList structure

- Removed ScrollView from MarketHomeV2, directly rendering MarketTokenList for improved layout.
- Updated MarketTokenList to use Stack for better responsiveness and scrolling behavior, ensuring a more flexible design.

limichange committed:
style: enhance scrolling behavior in MarketTokenList and add custom scrollbar styles

- Updated MarketTokenList to use a custom scrollbar for improved aesthetics and usability.
- Adjusted Stack component properties for better responsiveness and layout consistency.

limichange committed:
Merge branch 'x' into market-home-v2-ui

limichange committed:
refactor: update MarketHomeV2 and MarketTokenListNetworkSelector components

- Replaced the conditional rendering of headers in MarketHomeV2 with a TabPageHeader for improved navigation.
- Enhanced MarketTokenListNetworkSelector to utilize default enabled networks for better network selection experience.
- Removed unused imports and streamlined the component structure for clarity and maintainability.

limichange committed:
chore: remove MarketTokenListNetworkSelector data file

- Deleted the data.ts file containing network information for MarketTokenListNetworkSelector to streamline the component and eliminate unused code.

limichange committed:
refactor: simplify NetworksSearchPanel component

- Removed the unused swapNetworksIncludeAllNetwork import and adjusted the networksData assignment to directly use networksProp, enhancing clarity and reducing potential runtime issues.

limichange committed:
refactor: update MarketTokenListNetworkSelector to use presetNetworksMap

- Replaced the previous method of obtaining default enabled networks with a new function that utilizes presetNetworksMap for improved clarity and maintainability.
- Streamlined the component by removing unused imports and ensuring the network selection logic is more straightforward.

limichange committed:
feat: add navigation to MarketTokenList for detailed token view

- Introduced a new hook, useToDetailPage, to handle navigation to the token detail page.
- Updated MarketTokenList to utilize the new hook, allowing users to navigate to the Bitcoin detail page on item press.
- Enhanced the component's interactivity and user experience by integrating detailed token views.

limichange committed:
feat: integrate RiskIndicator and Txns components into MarketTokenList

- Added RiskIndicator and Txns components to enhance the MarketTokenList with risk assessment and transaction statistics.
- Updated MarketTokenData to reflect new audit types and removed the old Txns component.
- Refactored the rendering logic in MarketTokenColumns to utilize the new RiskIndicator component for displaying audit information.

limichange committed:
Merge branch 'x' into market-home-v2-ui

limichange committed:
feat: enhance MarketTokenList with TokenIdentityItem and address updates

- Introduced TokenIdentityItem component for improved token display in MarketTokenList.
- Updated MarketTokenColumns to render TokenIdentityItem, replacing the previous name display.
- Added address field to IMarketToken interface and populated it with placeholder addresses for existing tokens.
- Enhanced RiskIndicator component with styling adjustments for better visual integration.

limichange committed:
feat: update MarketTokenList to include token image URIs

- Enhanced MarketTokenColumns to render TokenIdentityItem with token and network logo URIs.
- Added tokenImageUri field to IMarketToken interface and populated it for existing tokens.
- Removed unused networkId prop from TokenIdentityItem for cleaner component structure.

limichange committed:
Merge branch 'x' into market-home-v2-ui

limichange committed:
Merge branch 'x' into market-home-v2-ui

sidmorizon committed:
Merge branch 'x' into market-home-v2-ui
